src/mrpro/operators/_CartesianSamplingOp.py

```diff
@@ -92,9 +93,9 @@ def forward(self, x: torch.Tensor) -> tuple[torch.Tensor,]:
         if not self._needs_indexing:
             return (x,)
 
-        x_2d = rearrange(x, '... k2_enc k1_enc k0_enc -> (...) (k2_enc k1_enc k0_enc)')
+        x_kflat = rearrange(x, '... coil k2_enc k1_enc k0_enc -> ... coil (k2_enc k1_enc k0_enc)')
         # take_along_dim does broadcast, so no need for extending here
-        x_indexed = torch.take_along_dim(x_2d, self._fft_idx, dim=-1)
+        x_indexed = torch.take_along_dim(x_kflat, self._fft_idx, dim=-1)
         # reshape to (... other coil, k2, k1, k0)
         x_reshaped = x_indexed.reshape(x.shape[:-3] + self._kshape[-3:])
```

Comment:
self._kshape is not really the shape of the k-space data but the shape of the trajectory. Should we rename it?

Reply:
Funny. I had the same idea initially but undid the change. 
 it _should_ be the shape of the k space, using the trajectory is just how we estimate the shape... 
So in the forward and adjoint it reads better with kspape...